src/features/Details/context/DetailsContext.tsx

```diff
@@ -0,0 +1,86 @@
+import { AppContext } from '@context/AppContext';
+import { Superhero } from '@interfaces/superhero.interface';
+import {
+  createContext,
+  FC,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+
+interface DetailsState {
+  areDetailsLoading: boolean;
+  superhero?: Superhero;
+  fetchDetails: (slug: string) => Promise<Superhero | undefined>;
+}
+
+export const detailsInitialState: DetailsState = {
+  areDetailsLoading: true,
+  superhero: undefined,
+  fetchDetails: () => Promise.resolve(undefined),
+};
+
+export const DetailsContext = createContext<DetailsState>(detailsInitialState);
+
+export const DetailsProvider: FC = ({ children }) => {
```

Comment:
**thought** I feel like this should not be a context but a hook named useHeroesDetails(), as well as the AppContext, should be a hook useHeroes(). 

**question** Why did you decide to use contexts over the hooks?

Reply:
> Why did you decide to use contexts over the hooks?

@kamil-borecki-cc Yes, now I realize it's kinda pointless ;) It should have been just a hook.